app/src/main/java/cs465/illinois/edu/dogonthequad/MapActivity.java

```diff
@@ -39,6 +45,53 @@ protected void onCreate(Bundle savedInstanceState) {
         MapFragment mapFragment = (MapFragment) getFragmentManager()
                 .findFragmentById(R.id.map);
         mapFragment.getMapAsync(this);
+
+        Button debugButton = findViewById(R.id.debug_button);
+
+        debugButton.setOnLongClickListener(new View.OnLongClickListener() {
+            int checkedItem = 0;
+
+            @Override
+            public boolean onLongClick(View view) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(MapActivity.this);
+                builder.setTitle("Sneaky Debug Menu");
+                List<String> names = new ArrayList<>();
+                try{
+                    names = API.getPresetFileNames(MapActivity.this);
+                } catch (IOException e){
+                    //do nothing
+                }
+                builder.setSingleChoiceItems(names.toArray(new String[0]), checkedItem, new DialogInterface.OnClickListener() {
```

Comment:
These OnClickListeners can be implemented as a lambda if you want. Then you don't have to define all of the stuff like:
```
new DialogInterface.OnClickListener() {
    @Override
    public void onClick(DialogInterface dialog, int which)
    ...
}
```